src/util/tile-error-collection.js

```diff
@@ -0,0 +1,77 @@
+var Backbone = require('backbone');
+var $ = require('jquery');
+var getValue = require('./get-object-value');
+var tileErrorImage = require('../../themes/img/tile-error.svg');
+
+var TILE_ERROR_IMAGE = 'data:image/svg+xml;base64,' + window.btoa(tileErrorImage);
+
+var TileErrorCollection = Backbone.Collection.extend({
+  initialize: function () {
+    this.running = false;
+    this.queue = new Backbone.Collection();
+
+    this.on('add', function (model) {
+      var isInQueue = this.queue.findWhere({ url: model.get('url') });
+      if (!isInQueue) this.queue.add(model);
+
+      model.get('node').src = TILE_ERROR_IMAGE;
+
+      if (!this.running) this._getTileErrors();
+    });
+  },
+
+  resetErrorTiles: function () {
+    var models = this.filter(function (model) {
```

Comment:
review this method, could we use proper "backbone" methods?

Reply:
I've been doing some research and I can only see `remove` and `reset`, the main difference is remove triggers a lot more events than reset